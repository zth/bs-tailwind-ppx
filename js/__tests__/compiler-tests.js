let { processCss, extractNodeData } = require("../utils");

async function parseAndGetFirstNode(src) {
  let parsed = await processCss(src);
  return parsed.root.nodes[0];
}

describe("Compiler tests", () => {
  describe("processCss", () => {
    it("processes the CSS and returns the result", async () => {
      let css = await processCss(`
            .bg-black-100 {
                background-color: black;
            }
            `);

      expect(css.css).toMatchSnapshot();
    });
  });

  describe("extractNodeData", () => {
    async function extractNode(src) {
      return extractNodeData(await parseAndGetFirstNode(src));
    }

    it("extracts regular class names", async () => {
      expect(
        await extractNode(".bg-black-100 { background-color: black }")
      ).toBe("bg-black-100");
    });

    it("ignores anything with more than 1 class selector", async () => {
      expect(
        await extractNode(".bg-black-100.bg-hello { background-color: black }")
      ).toBeUndefined();
    });

    it("cleans escaped characters", async () => {
      expect(
        await extractNode(".focus\\:bg-black-100 { background-color: black }")
      ).toBe("focus:bg-black-100");
    });

    it("cleans pseudo selectors", async () => {
      expect(
        await extractNode(
          ".focus\\:bg-black-100:focus { background-color: black }"
        )
      ).toBe("focus:bg-black-100");
      expect(
        await extractNode(
          ".focus\\:bg-black-100:hover { background-color: black }"
        )
      ).toBe("focus:bg-black-100");
      expect(
        await extractNode(".bg-black-100::after { background-color: black }")
      ).toBe("bg-black-100");
    });
  });
});
