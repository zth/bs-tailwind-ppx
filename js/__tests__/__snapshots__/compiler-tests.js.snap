// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Compiler tests processCss processes the CSS and returns the result 1`] = `
".bg-black-100 {
  background-color: black;
}
            "
`;
